### PR TITLE
Edit README

### LINKED ISSUE #1936: pyMOR features in README
The pyMOR intro ("pyMOR is a software library for...") should be updated (it's no longer only providing RB and sys-mor methods). Perhaps there should be a whole section in the README listing pyMOR features, and explaining the typical use cases, cutting edge examples, and things out of reach or to be done in the future.

### PR DESCRIPTION
**Pull request by pmli**
This PR is to close #1936 and #1934 and fix some other issues:

- [x] add "features" section
- [x] link to MOR Wiki
- [x] use `#` for section formatting
- [x] remove license text and link to the license file (to avoid the screaming font)
- [x] restructure the "installation" section (such that the pip section doesn't look long and complicated)
- [x] recommend using mamba instead of conda (mamba resolves much quicker)
- [x] remove the "useful links" section
- [x] remove the "external matrix equations solvers" section (because they are also mentioned in the "installation" section)
- [x] remove reference to `pymor.discretizers.disk`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -124,36 +112,35 @@ so you might prefer to install pyMOR from the current release branch:
 Release branches will always stay stable and will only receive bugfix commits
 after the corresponding release has been made.
 
-Installation via conda
-----------------------
+### conda
```
Comment (ftschindler) [inline README.md:115]:
```suggestion
### conda/mamba
```

File: README.md [hunk 2]
```diff
@@ -124,36 +112,35 @@ so you might prefer to install pyMOR from the current release branch:
...
+
+We recommend installation of pyMOR in a
+[conda environment](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html).
```
Comment (ftschindler) [inline README.md:118]:
```suggestion
[conda environment](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html)
with mamba.
```

**Review by ftschindler — APPROVED** [review]
Gave you an optional suggestion to highlight mamba a bit more prominently.

**Comment by HenKlei** [comment]
Do you think the "features" section is detailed enough, @pmli? In my opinion, this could be extended quite a bit.

**Comment by pmli** [comment]
> Do you think the "features" section is detailed enough, @pmli? In my opinion, this could be extended quite a bit.

I think it can and definitely should be extended significantly. I should have said this in the first comment, but I was wondering what people think about shortening the pyMOR description (the part between `# pyMOR` and `## Features`) to just one sentence.

**Comment by sdrave** [comment]
Looks good to me. I have updated the section on external solvers an added a comment regarding minimal dependencies for the demos.

What I noticed: We don't mention the main-developers email address. Should we do that? For some topics, Github discussions might not be adequate (like offering us a huge grant or so). So it might be good to offer a more formal way of contacting us.

**Comment by pmli** [comment]
> Looks good to me. I have updated the section on external solvers an added a comment regarding minimal dependencies for the demos.

Thanks.

> What I noticed: We don't mention the main-developers email address. Should we do that? For some topics, Github discussions might not be adequate (like offering us a huge grant or so). So it might be good to offer a more formal way of contacting us.

I also think Discussions are not exactly a place for direct contact. Is `main.developers@pymor.org` still active?
We do have the main devs email addresses in `AUTHORS.md`, so we could link to that.

**Comment by sdrave** [comment]
> I also think Discussions are not exactly a place for direct contact. Is `main.developers@pymor.org` still active? We do have the main devs email addresses in `AUTHORS.md`, so we could link to that.

Yes it does ...

**Comment by sdrave** [comment]
Wow, I have never seen this `pip` incantation with `@`. When was that introduced?
Regarding 'useful links', maybe we can remove all of them? I don't think that just linking to `requirements.txt` is useful. The other links can be easily found from the docs.
I also think that the 'Matrix Equation Solvers' section can be entirely removed. (If someone feels the need, one can add one sentence or two to the installation section.

**Comment by pmli** [comment]
> Wow, I have never seen this `pip` incantation with `@`. When was that introduced?

Not sure. I wanted to try the "egg" version with the recent `pip` because using eggs seems outdated, and then `pip` complained that installing that way will be deprecated in version 25.

> Regarding 'useful links', maybe we can remove all of them? I don't think that just linking to `requirements.txt` is useful. The other links can be easily found from the docs.

I agree about removing the link to `requirements.txt`, maybe even the release notes, but I think it might be nice to give links to "getting started" and the tutorials (similar to developer docs). Maybe what is currently in the "Documentation" section can be moved to developer docs.

> I also think that the 'Matrix Equation Solvers' section can be entirely removed. (If someone feels the need, one can add one sentence or two to the installation section.

Ok, I'll work on that.

File: README.md [hunk 3]
```diff
@@ -186,35 +171,19 @@ we provide bindings for the following solver libraries:
...
-which are needed for the system-theoretic methods and need to be installed
-separately. Bindings for the following solver libraries are included:
-
-* [Py-M.E.S.S.](https://www.mpi-magdeburg.mpg.de/projects/mess)
-
-    The Matrix Equation Sparse Solver library is intended for solving large sparse matrix equations (`pymor.bindings.pymess`).
-
-* [Slycot](https://github.com/python-control/Slycot)
+A simple example for direct integration of pyMOR with a a custom solver
+can be found in `src/pymordemos/minimal_cpp_demo`.
```
Comment (HenKlei) [inline README.md:175]:
```suggestion
can be found in `pymordemos.minimal_cpp_demo`.
```

Comment (HenKlei) [reply README.md:175]: Just to use the same notation as above.

File: README.md [hunk 4]
```diff
@@ -23,149 +9,148 @@ provided for getting started quickly.
...
-THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+* Reduced basis methods for parametric linear and non-linear problems.
+* System-theoretic methods for linear time-invariant systems.
+* Neural network-based methods for parametric problems.
+* Proper orthogonal decomposition.
+* Dynamic mode decomposition.
+* Rational interpolation of data (Loewner, AAA).
+* Numerical linear algebra (Gram-Schmidt, time-stepping, ...).
+* Pure Python implementations of finite element and finite volume
+  discretizations using the NumPy/SciPy scientific computing stack.
```
Comment (HenKlei) [inline README.md:31]: Do we want to be more specific here? For instance, one could mention which system-theoretic methods are actually implemented or that things like greedy, empirical interpolation and error estimation are available... However, that might also be too much, I am not sure.

File: README.md [hunk 5]
```diff
@@ -23,149 +9,148 @@ provided for getting started quickly.
...
---------------------
+## Installation via pip
 
 We recommend installation of pyMOR in a [virtual environment](https://virtualenv.pypa.io/en/latest/).
 
 pyMOR can easily be installed with the [pip](https://pip.pypa.io/en/stable/)
-command:
+command.
+Please note that pip versions prior to 21.1 might have problems resolving all
+dependencies, so running the following first is recommended.
```
Comment (HenKlei) [inline README.md:54]:
```suggestion
dependencies, so running the following first is recommended:
```

File: README.md [hunk 6]
```diff
@@ -23,149 +9,148 @@ provided for getting started quickly.
...
     pip install pymor
 
-for an installation with minimal dependencies.
+Note that most included demo scripts additionally require `matplotlib` and
+Qt bindings such as `pyside2` to function.
+
+### Latest Release (with all Optional Dependencies)
+
+The following installs the latest release of pyMOR on your system with most
+optional dependencies.
```
Comment (HenKlei) [inline README.md:75]:
```suggestion
optional dependencies:
```

File: README.md [hunk 7]
```diff
@@ -23,149 +9,148 @@ provided for getting started quickly.
...
+Documentation is available [online](https://docs.pymor.org/).
+We recommend starting with
+[getting started](https://docs.pymor.org/latest/getting_started.html),
+[tutorials](https://docs.pymor.org/latest/tutorials.html), and
+[technical overview](https://docs.pymor.org/latest/technical_overview.html).
 
-Useful Links
-------------
+To build the documentation locally,
+run the following from inside the root directory of the pyMOR source tree.
```
Comment (HenKlei) [inline README.md:143]:
```suggestion
run the following from inside the root directory of the pyMOR source tree:
```

File: README.md (hunk 4 above)
Comment (sdrave) [reply README.md:31]: Maybe let's discuss this elsewhere and first merge this pr, @pmli, @HenKlei?

**Review by sdrave — APPROVED** [review]

**Review by ftschindler — APPROVED** [review]
Final readme looks good!

File: README.md (hunk 4 above)
Comment (pmli) [reply README.md:31]: Fine with me.

**Pull request merged**